src/vs/workbench/contrib/tasks/browser/taskTerminalStatus.ts

```diff
@@ -52,11 +52,22 @@ export class TaskTerminalStatus extends Disposable {
 		}));
 	}
 
+	public override dispose(): void {
+		super.dispose();
+		for (const terminalData of this.terminalMap.values()) {
+			terminalData.disposeListener?.dispose();
+		}
+		this.terminalMap.clear();
+	}
+
```

Comment:
I don't like overriding dispose as the compiler doesn't force you to call the super. Can you instead add this to the ctor?

```ts
this._register(toDisposable(() => {
	for (const terminalData of this.terminalMap.values()) {
		terminalData.disposeListener?.dispose();
	}
	this.terminalMap.clear();
});
```

Also `terminalData.disposeListener` might be better as a `MutableDisposable`